test_stopwatch.py

```diff
@@ -47,6 +47,29 @@ def test_default_exports(self):
         sw = StopWatch()
         add_timers(sw)
 
+    def test_stopwatch_cancel(self):
+        """Test that spans can be correctly cancelled and not reported."""
+        sw = StopWatch()
+        sw.start('root')
+        sw.start('child')
+        sw.cancel('child')
+        sw.end('root')
+        agg_values = sw.get_last_aggregated_report().aggregated_values
+        assert len(agg_values) == 1
+        assert 'root' in agg_values
+
+    def test_stopwatch_cancel_context_manager(self):
+        """Test that spans can be cancelled while inside a span context."""
+        sw = StopWatch()
+        with sw.timer('root'):
+            with sw.timer('child'):
+                sw.cancel('child')
+                with sw.timer('grand'):
+                    pass
+        agg_values = sw.get_last_aggregated_report().aggregated_values
+        assert len(agg_values) == 2
+        assert all([span in agg_values for span in ('root', 'root#grand')])
```

Comment:
It looks like there isn't a normal test which creates multiple root spans, but I think this is a behavior we expect to work. I believe the client/server already uses it that way.

Might want to add a test for that as well. Something along these lines (with asertions / boilerplate added.
```
def test_multiple_roots(self):
    with sw.timer('root'):
        pass
    with sw.timer('root'):
        passs
```